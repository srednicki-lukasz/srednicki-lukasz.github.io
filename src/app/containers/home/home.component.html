<div class="h-100 card">
  <div class="card-body d-flex flex-column align-items-center justify-content-between px-0">
    <section class="flex-fill container-fluid p-lg-5">
      <div class="h-100 row">
        <div class="text-section col-12 col-lg-6">
          <div class="d-lg-none">
            <h1 class="text-center">
              {{ authorInfo.name }}
              <br />
              {{ authorInfo.surname }}
            </h1>

            <hr class="w-100 text-secondary" />

            <h6 class="text-center">{{ authorInfo.title }}</h6>
          </div>

          <div class="d-none d-lg-block">
            <h1>
              <span class="ps-1">{{ authorInfo.name }}</span>
              <br />
              {{ authorInfo.surname }}
            </h1>

            <hr class="w-100 text-secondary" />

            <h6>
              <span class="accent">>_</span>
              {{ authorInfo.title }}
            </h6>
          </div>
        </div>

        <div class="image-section d-none d-lg-block col-12 col-lg-6">
          <img src="balloon.svg" class="img-fluid" alt="" />
        </div>
      </div>
    </section>

    <div class="w-100 carousel">
      <div class="d-flex align-items-center">
        <div class="carousel-track">
          <div class="w-100 d-inline-block carousel-items">
            @for (logo of carouselItems; track $index) {
              <img height="40" [src]="logo.src" [alt]="logo.alt" />
            }
          </div>

          <div class="w-100 d-inline-block carousel-items">
            @for (logo of carouselItems; track $index) {
              <img height="40" [src]="logo.src" [alt]="logo.alt" />
            }
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
