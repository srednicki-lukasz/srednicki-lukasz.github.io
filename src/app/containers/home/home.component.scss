@import '../../../styles.scss';

:host {
  @include router-base;
  padding: 12px;
}

.card {
  color: var(--color);
  overflow: hidden;
  box-shadow: var(--card-box-shadow);
  background-color: var(--card-background);
  background-image: url('../../../../public/texture.svg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center top;

  .text-section {
    align-content: center;
    line-height: 3rem;

    h1 {
      font-size: clamp(2.5em, 10vw, 5rem);
    }

    h6 {
      font-size: clamp(1em, 2vw, 2rem);
    }
  }

  .image-section {
    align-content: center;

    img {
      animation: float 5s ease-in-out infinite;
    }
  }
}

@keyframes float {
  0% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(-20px);
  }
  100% {
    transform: translateY(0);
  }
}

.carousel {
  overflow: hidden;
  position: relative;

  &::before,
  &::after {
    content: '';
    position: absolute;
    top: 0;
    width: 10%;
    height: 100%;
    z-index: 1;
    pointer-events: none;
  }

  &::before {
    left: 0;
    background: linear-gradient(to right, var(--card-background), rgba(255, 255, 255, 0));
  }

  &::after {
    right: 0;
    background: linear-gradient(to left, var(--card-background), rgba(255, 255, 255, 0));
  }

  .carousel-track {
    white-space: nowrap;
    animation: slide-carousel 40s linear infinite;

    &:hover {
      animation-play-state: paused;
    }

    .carousel-items > img {
      width: calc(100% / 8);
      padding: 0 30px;
    }
  }
}

@keyframes slide-carousel {
  from {
    transform: translate3d(0, 0, 0);
  }
  to {
    transform: translate3d(-100%, 0, 0);
  }
}
