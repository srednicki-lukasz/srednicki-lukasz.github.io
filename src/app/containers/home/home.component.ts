import { ChangeDetectionStrategy, Component } from '@angular/core';
import { authorInfo } from '../../author-info';

@Component({
  selector: 'home',
  templateUrl: './home.component.html',
  styleUrl: './home.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HomeComponent {
  readonly authorInfo = authorInfo;

  readonly carouselItems = [
    { src: 'carousel-images/html.svg', alt: 'HTML logo' },
    { src: 'carousel-images/css.svg', alt: 'CSS logo' },
    { src: 'carousel-images/javascript.svg', alt: 'JavaScript logo' },
    { src: 'carousel-images/typescript.svg', alt: 'TypeScript logo' },
    { src: 'carousel-images/angular.svg', alt: 'Angular logo' },
    { src: 'carousel-images/ngrx.svg', alt: 'NgRx logo' },
    { src: 'carousel-images/rxjs.svg', alt: 'RxJS logo' },
    { src: 'carousel-images/python.svg', alt: 'Python logo' },
  ];
}
