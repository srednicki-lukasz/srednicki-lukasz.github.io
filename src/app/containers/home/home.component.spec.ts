import { importProvidersFrom } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { HomeComponent } from './home.component';

describe('HomeComponent', () => {
  let spectator: Spectator<HomeComponent>;

  const createComponent = createComponentFactory({
    component: HomeComponent,
    providers: [importProvidersFrom(RouterTestingModule)],
    detectChanges: false,
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should match snapshot', () => {
    spectator.detectChanges();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
