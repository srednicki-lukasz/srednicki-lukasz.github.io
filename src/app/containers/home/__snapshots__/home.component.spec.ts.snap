// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should match snapshot 1`] = `
<home
  authorInfo={[Function Object]}
  carouselItems={[Function Array]}
>
  <div
    class="h-100 card"
  >
    <div
      class="card-body d-flex flex-column align-items-center justify-content-between px-0"
    >
      <section
        class="flex-fill container-fluid p-lg-5"
      >
        <div
          class="h-100 row"
        >
          <div
            class="text-section col-12 col-lg-6"
          >
            <div
              class="d-lg-none"
            >
              <h1
                class="text-center"
              >
                 Łukasz 
                <br />
                 Średnicki 
              </h1>
              <hr
                class="w-100 text-secondary"
              />
              <h6
                class="text-center"
              >
                Front-End Developer
              </h6>
            </div>
            <div
              class="d-none d-lg-block"
            >
              <h1>
                <span
                  class="ps-1"
                >
                  Łukasz
                </span>
                <br />
                 Średnicki 
              </h1>
              <hr
                class="w-100 text-secondary"
              />
              <h6>
                <span
                  class="accent"
                >
                  &gt;_
                </span>
                 Front-End Developer 
              </h6>
            </div>
          </div>
          <div
            class="image-section d-none d-lg-block col-12 col-lg-6"
          >
            <img
              alt=""
              class="img-fluid"
              src="balloon.svg"
            />
          </div>
        </div>
      </section>
      <div
        class="w-100 carousel"
      >
        <div
          class="d-flex align-items-center"
        >
          <div
            class="carousel-track"
          >
            <div
              class="w-100 d-inline-block carousel-items"
            >
              <img
                alt="HTML logo"
                height="40"
                src="carousel-images/html.svg"
              />
              <img
                alt="CSS logo"
                height="40"
                src="carousel-images/css.svg"
              />
              <img
                alt="JavaScript logo"
                height="40"
                src="carousel-images/javascript.svg"
              />
              <img
                alt="TypeScript logo"
                height="40"
                src="carousel-images/typescript.svg"
              />
              <img
                alt="Angular logo"
                height="40"
                src="carousel-images/angular.svg"
              />
              <img
                alt="NgRx logo"
                height="40"
                src="carousel-images/ngrx.svg"
              />
              <img
                alt="RxJS logo"
                height="40"
                src="carousel-images/rxjs.svg"
              />
              <img
                alt="Python logo"
                height="40"
                src="carousel-images/python.svg"
              />
            </div>
            <div
              class="w-100 d-inline-block carousel-items"
            >
              <img
                alt="HTML logo"
                height="40"
                src="carousel-images/html.svg"
              />
              <img
                alt="CSS logo"
                height="40"
                src="carousel-images/css.svg"
              />
              <img
                alt="JavaScript logo"
                height="40"
                src="carousel-images/javascript.svg"
              />
              <img
                alt="TypeScript logo"
                height="40"
                src="carousel-images/typescript.svg"
              />
              <img
                alt="Angular logo"
                height="40"
                src="carousel-images/angular.svg"
              />
              <img
                alt="NgRx logo"
                height="40"
                src="carousel-images/ngrx.svg"
              />
              <img
                alt="RxJS logo"
                height="40"
                src="carousel-images/rxjs.svg"
              />
              <img
                alt="Python logo"
                height="40"
                src="carousel-images/python.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</home>
`;
